### ISSUE TITLE
Problem #28: U and V are non-unique with respect to signs

### ISSUE THREAD

**Issue by Open-Deep-ML** [Open-Deep-ML/DML-OpenProblem#14]
U and V are non-unique with respect to signs, i.e. multiplying -1 to row i of U and column i of V^T will give us the same matrix.
Source: https://x.com/12345pnp/status/1819301249830772923
https://github.com/Open-Deep-ML/DML-OpenProblem/blob/3fc961c1518fa794e0689cf298c0fa58a33dce5a/Problems/28_svd_2x2_eign/solution.py#L1

**Comment by Open-Deep-ML** [comment]
Seems like the solution I have is the same as numpy except numpy S and D are multiplied by -1
Numpy:
SVDResult(
S=array([[-0.8,  0.6], [ 0.6,  0.8]]), 
V=array([15.65247584,  4.47213595]),
D=array([[ 0.89442719, -0.4472136 ],
       [ 0.4472136 ,  0.89442719]]))

Deep-ML
S = (array([[-0.8, -0.6],
        [ 0.6, -0.8]]),
 V = array([[15.65247584,  0.        ],
        [ 0.        ,  4.47213595]]),
D =  array([[ 0.89442719, -0.4472136 ],
        [-0.4472136 , -0.89442719]]))

Seems good to me, But I may be miss understanding the issue

**Comment by 12345pnp** [comment]
The issue is the numpy result is being considered invalid.

![image](https://github.com/user-attachments/assets/6f7b5023-5587-4f05-999a-bba8a03d28de)

**Comment by moe18** [comment]
Would you be able to add the code you tried to submit?

**Comment by EpsIotaPi** [comment]
I think the solution of this question is not correct, the expected result did not satisfy A = U @ S @ V^T

```
print(svd_2x2(np.array([[-10, 8], [10, -1]])))
# (array([[  0.8, -0.6], [-0.6, -0.8]]), array([15.65247584,  4.47213595]), array([[ -0.89442719,  0.4472136], [ -0.4472136 , -0.89442719]]))


U, S, V = (np.array([[  0.8, -0.6], [-0.6, -0.8]]), np.array([15.65247584,  4.47213595]), np.array([[ -0.89442719,  0.4472136], [ -0.4472136 , -0.89442719]]))
print(U @ np.diag(S) @ V.transpose())
# [[-12.4         -3.20000006]
#  [  6.79999997   7.40000003]]
```

I will provide my solution and the test method later.

**Comment by moe18** [comment]
the solution for this problem, is made to replicate the method used within np.linalg.svd, which also does not compute the correct reconstruction, I think the best course of action is to create a new question for this method of solving SVD (so far, we have 2 questions about SVD and this will be out third so will need to think of a good naming convention)

![image](https://github.com/user-attachments/assets/1a2c9a3a-48d9-4046-91b5-50a9d82a9864)

**Comment by moe18** [comment]
@EpsIotaPi, comment in the new issue I opened if you would like to create the new problem (you pretty much already made it with your last PR it's just a matter of reformatting it as a new problem)

**Comment by EpsIotaPi** [comment]
Your code is not quite correct, np’s implementation returns V which is transposed. So you don’t need to transpose it again.
You could try with the following code:

```
A = np.array([[-10, 8], [10, -1]])

U, s, V = np.linalg.svd(A)

print(U @ np.diag(s) @ V)

# [[-10.   8.]
#  [ 10.  -1.]]
```

<img width="1367" alt="Screenshot 2024-08-05 at 19 41 28" src="https://github.com/user-attachments/assets/41db34d4-fe0d-4c44-925d-71a63a270eef">


> the solution for this problem, is made to replicate the method used within np.linalg.svd, which also does not compute the correct reconstruction, I think the best course of action is to create a new question for this method of solving SVD (so far, we have 2 questions about SVD and this will be our third so will need to think of a good naming convention)
> 
> ![image](https://private-user-images.githubusercontent.com/26696833/355083072-1a2c9a3a-48d9-4046-91b5-50a9d82a9864.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI4NTgwOTcsIm5iZiI6MTcyMjg1Nzc5NywicGF0aCI6Ii8yNjY5NjgzMy8zNTUwODMwNzItMWEyYzlhM2EtNDhkOS00MDQ2LTkxYjUtNTBhOWQ4MmE5ODY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA1VDExMzYzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiNzE2MmEyMGQ1MGEzMWQ5YTZiZTFiMWM4ZTExYzUwYmE5OWIxM2ZiMTQzOTVhYmRkNDU2MWIyNzkxNzZjNjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QKjI5MUGP-HskJ8KvClBvGSViFX9IBU-PMT1QHU5ORQ)

**Comment by EpsIotaPi** [comment]
Oh, I just realized that the original method also provide the transposed matrix V ......
I think my method is quite similar to yours, so we does not need to open a new problem, instead of the following revised:

1.  Decide whether the function should return V or V.T and clarify it in the description  of problem 28.
2. Revise the `test_svd_2x2()`. I suggest use the method I wrote in PR, which could accept both both negative and non-negative results.
